File: probs/5052.py
import sys
input = sys.stdin.readline

input = open('input.txt', 'r').readline

for _ in range(int(input())):
    N = int(input())
    nums = [input().strip() for _ in range(N)]
    nums.sort()

    for i in range(N-1):
        if len(nums[i]) < len(nums[i+1]) and nums[i+1][:len(nums[i])] == nums[i]:
            print('NO')
            break
    else:
        print('YES')

File: probs/5052.cpp
#include <bits/stdc++.h>

using namespace std;

vector<string> S;

int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    freopen("input.txt", "r", stdin);

    int T; cin >> T;
    while (T--) {
        int N; cin >> N;
        S.clear(); S.resize(N);
        for (int i = 0; i < N; i++) cin >> S[i];
        sort(S.begin(), S.end());

        int ans = 1;
        for (int i = 1; i < N; i++)
            ans &= S[i].find(S[i-1]) != 0;
        cout << (ans ? "YES" : "NO") << '\n';
    }

    return 0;
}
